### PR TITLE
Chapter 3

### PR DESCRIPTION
**Pull request by starocketeer**
MEKDI

### CODE REVIEW

**Comment by starocketeer** [comment]
Ready

File: .gitignore [hunk 1]
```diff
@@ -1,56 +1,45 @@
...
+/public/packs
+/public/packs-test
+/node_modules
+/yarn-error.log
+yarn-debug.log*
+.yarn-integrity
+/config/credentials.yml.enc
+/config/database.yml
+/test
+/app/assets/stylesheets/static_pages.scss
```
Comment (saobv-2223) [inline .gitignore]: add endline

File: app/assets/stylesheets/application.css [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+ * You're free to add application-wide styles to this file and they'll appear at the bottom of the
+ * compiled file so the styles you add here take precedence over styles defined in any other CSS/SCSS
+ * files in this directory. Styles in this file should be added after the last require_* statement.
+ * It is generally better to create a new file per style scope.
+ *
+ *= require_tree .
+ *= require_self
+ */
+@import 'bootstrap - sprockets';
+@import 'bootstrap';
```
Comment (saobv-2223) [inline app/assets/stylesheets/application.css]: add endline

File: app/assets/stylesheets/color.scss [hunk 3]
```diff
@@ -0,0 +1,2 @@
+$blue: #0006FF;
+$red: #F90E05;
```
Comment (saobv-2223) [inline app/assets/stylesheets/color.scss]: add endline

File: app/assets/stylesheets/custom.scss [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+  
+  textarea {
+    resize: vertical;
+  }
+  
+  .center {
+    text-align: center;
+  }
+  
+  .center h1 {
```
Comment (saobv-2223) [inline app/assets/stylesheets/custom.scss]: remove 2 spaces

File: app/assets/stylesheets/custom.scss [hunk 5]
```diff
@@ -0,0 +1,41 @@
...
+    letter-spacing: -1px;
+    padding-top: 9px;
+    font-weight: bold;
+  }
+  
+  #logo:hover {
+    color: $red;
+    text-decoration: none;
+  }
+
```
Comment (saobv-2223) [inline app/assets/stylesheets/custom.scss]: add endline

File: app/controllers/static_pages_controller.rb [hunk 6]
```diff
@@ -0,0 +1,5 @@
+class StaticPagesController < ApplicationController
+    def home; end
+    
+    def help; end
+  end
```
Comment (saobv-2223) [inline app/controllers/static_pages_controller.rb]: indenting 2 spaces and add endline

**Review by saobv-2223 — COMMENTED** [review]
Please, follow this topic and fix all mistakes
https://docs.google.com/document/d/17CBB-BCQm6wu60cCiAmYf7tibU-VaKLe1TytCU9wSVc/edit?usp=sharing

File: .gitignore (hunk 1 above)
Comment (starocketeer) [reply .gitignore]: done

File: app/assets/stylesheets/application.css (hunk 2 above)
Comment (starocketeer) [reply app/assets/stylesheets/application.css]: done

File: app/controllers/static_pages_controller.rb (hunk 6 above)
Comment (starocketeer) [reply app/controllers/static_pages_controller.rb]: done

File: app/assets/stylesheets/custom.scss (hunk 4 above)
Comment (starocketeer) [reply app/assets/stylesheets/custom.scss]: Done

File: app/assets/stylesheets/color.scss (hunk 3 above)
Comment (starocketeer) [reply app/assets/stylesheets/color.scss]: Done

File: app/assets/stylesheets/custom.scss (hunk 5 above)
Comment (starocketeer) [reply app/assets/stylesheets/custom.scss]: Ready

**Comment by starocketeer** [comment]
Ready

File: app/assets/stylesheets/custom.scss [hunk 7]
```diff
@@ -0,0 +1,32 @@
...
+    padding-top: 9px;
+    font-weight: bold;
+  }
+  
+  #logo:hover {
+    color: $red;
+    text-decoration: none;
+ }
+  
+
```
Comment (saobv-2223) [inline app/assets/stylesheets/custom.scss]:
```suggestion
 }

```

File: app/helpers/application_helper.rb [hunk 8]
```diff
@@ -1,2 +1,10 @@
 module ApplicationHelper
+<<<<<<< HEAD
```
Comment (saobv-2223) [inline app/helpers/application_helper.rb]: resolve conflict, please

**Review by saobv-2223 — COMMENTED** [review]
Resolve all conflict files, please

File: app/assets/stylesheets/custom.scss (hunk 7 above)
Comment (starocketeer) [reply app/assets/stylesheets/custom.scss]: done

File: app/helpers/application_helper.rb (hunk 8 above)
Comment (starocketeer) [reply app/helpers/application_helper.rb]: done

**Comment by starocketeer** [comment]
Ready

File: app/views/layouts/application.html.erb [hunk 9]
```diff
@@ -1,7 +1,11 @@
 <!DOCTYPE html>
 <html>
   <head>
+<<<<<<< HEAD
```
Comment (ngocvt-0484) [inline app/views/layouts/application.html.erb]: fix conflict please

File: app/views/layouts/mailer.text.erb [hunk 10]
```diff
@@ -1 +1,5 @@
+<<<<<<< HEAD
```
Comment (ngocvt-0484) [inline app/views/layouts/mailer.text.erb]: conflict

**Comment by ngocvt-0484** [comment]
please change 2 commits to 1 commit
```
git reset --soft origin/master
git add .
git commit -m "chapter 3"
```

**Comment by starocketeer** [comment]
Ready

File: app/views/layouts/application.html.erb (hunk 9 above)
Comment (ngocvt-0484) [reply app/views/layouts/application.html.erb]: not fix yet

Comment (starocketeer) [reply app/views/layouts/application.html.erb:4]: can you help me? I thought it is solved. What should I do to solve?

**Comment by starocketeer** [comment]
Ready

File: app/assets/stylesheets/custom.scss [hunk 11]
```diff
@@ -0,0 +1,36 @@
...
+  #logo:hover {
+    color: $red;
+    text-decoration: none;
+ }
+<<<<<<< HEAD
+   
+=======
+  
+ 
+>>>>>>> 28df35c... Chapter 3
```
Comment (saobv-2223) [inline app/assets/stylesheets/custom.scss]: remove all the conflict file, plese

**Review by saobv-2223 — COMMENTED** [review]
Resolve all the conflict file, please

File: app/assets/stylesheets/custom.scss (hunk 11 above)
Comment (starocketeer) [reply app/assets/stylesheets/custom.scss]: done

**Comment by starocketeer** [comment]
ready

File: app/helpers/application_helper.rb (hunk 8 above)
Comment (ngocvt-0484) [reply app/helpers/application_helper.rb]: still conflict
please remove all 
```
<<<<<<< HEAD
=======
>>>>>>> 
```
in your app

**Comment by starocketeer** [comment]
ready

**Comment by ngocvt-0484** [comment]
still conflict
please remove all
```
<<<<<<< HEAD
=======
>>>>>>> 
```
in your app

**Comment by ngocvt-0484** [comment]
commented

**Comment by starocketeer** [comment]
ready

File: app/views/static_pages/help.html.erb [hunk 12]
```diff
@@ -0,0 +1,9 @@
+<%= t provide :title, "Help" %>
+
+<h1>Help</h1>
+<p>
+  Get help on the Ruby on Rails Tutorial at the
+  <%= link_to "Rails Tutorial help page", Settings.links.rails_tutorial.help_page %>.
+  To get help on this sample app, see the
+  <%= link_to "<em>Ruby on Rails Tutorial</em>book", Settings.links.rails_tutorial.book_page %>.
+</p>
```
Comment (saobv-2223) [inline app/views/static_pages/help.html.erb]:
```suggestion
</p>

```

File: app/views/static_pages/home.html.erb [hunk 13]
```diff
@@ -0,0 +1,8 @@
+<%= t provide :title, "Home" %>
+
+<h1>Sample app</h1>
+<p>
+  This is the home page for the 
+  <%= link_to "Ruby on Rails Tutorial", Settings.links.rails_tutorial.home_page %> 
+  sample application.
+</p>
```
Comment (saobv-2223) [inline app/views/static_pages/home.html.erb]:
```suggestion
</p>

```

File: bebel.config.js [hunk 14]
```diff
@@ -0,0 +1,72 @@
...
+        ],
+        [
+          '@babel/plugin-transform-regenerator',
+          {
+            async: false
+          }
+        ]
+      ].filter(Boolean)
+    }
+  }
```
Comment (saobv-2223) [inline bebel.config.js]:
```suggestion
  }
  
```

File: config/credentials.yml.enc.example [hunk 15]
```diff
@@ -1 +1,5 @@
+<<<<<<< HEAD
```
Comment (saobv-2223) [inline config/credentials.yml.enc.example]: remove conflict

File: config/settings.yml [hunk 16]
```diff
@@ -0,0 +1,5 @@
+links:
+  rails_tutorial:
+    help_page: http://www.railstutorial.org/help
+    book_page: https://www.railstutorial.org/book
+    home_page: http://www.railstutorial.org/
```
Comment (saobv-2223) [inline config/settings.yml]:
```suggestion
    home_page: http://www.railstutorial.org/
    
```

File: config/webpacker.yml [hunk 17]
```diff
@@ -71,9 +71,15 @@ development:
     quiet: false
     pretty: false
     headers:
+<<<<<<< HEAD
```
Comment (saobv-2223) [inline config/webpacker.yml]: resolve conflict

File: app/controllers/static_pages_controller.rb [hunk 18]
```diff
@@ -0,0 +1,4 @@
+class StaticPagesController < ApplicationController
+    def home; end    
+    def help; end
+  end
```
[the remark this answers is not in the document]
Comment (starocketeer) [reply app/controllers/static_pages_controller.rb]: sorry, the comment says "Uploading image.png…" . I cannot see what need to be fix

File: app/helpers/application_helper.rb [hunk 19]
```diff
@@ -1,2 +1,6 @@
 module ApplicationHelper
-end
+    def full_title page_title
+        base_title = "Ruby on Rails Tutorial Sample App"
+        page_title.blank? ? base_title : "#{page_title} | #{base_title}"
+      end
+    end
```
[the remark this answers is not in the document]
Comment (starocketeer) [reply app/helpers/application_helper.rb]: sorry, the comment says "Uploading image.png…" . I cannot see what need to be fix

File: app/views/static_pages/help.html.erb (hunk 12 above)
Comment (starocketeer) [reply app/views/static_pages/help.html.erb]: Done

File: app/views/static_pages/home.html.erb (hunk 13 above)
Comment (starocketeer) [reply app/views/static_pages/home.html.erb]: Done

File: app/controllers/static_pages_controller.rb (hunk 18 above)
[the remark this answers is not in the document]
Comment (starocketeer) [reply app/controllers/static_pages_controller.rb]: done

File: app/helpers/application_helper.rb (hunk 19 above)
[the remark this answers is not in the document]
Comment (starocketeer) [reply app/helpers/application_helper.rb]: done

File: bebel.config.js (hunk 14 above)
Comment (starocketeer) [reply bebel.config.js]: done

File: config/credentials.yml.enc.example (hunk 15 above)
Comment (starocketeer) [reply config/credentials.yml.enc.example]: Done

File: config/settings.yml (hunk 16 above)
Comment (starocketeer) [reply config/settings.yml]: Done

File: config/webpacker.yml (hunk 17 above)
Comment (starocketeer) [reply config/webpacker.yml]: done

**Comment by starocketeer** [comment]
Ready

File: config/webpacker.yml [hunk 20]
```diff
@@ -71,9 +71,15 @@ development:
...
       'Access-Control-Allow-Origin': '*'
     watch_options:
       ignored: '**/node_modules/**'
+=======
+      "Access-Control-Allow-Origin": "*"
+    watch_options:
+      ignored: "**/node_modules/**"
+>>>>>>> a07d1d6... Init Project
```
Comment (saobv-2223) [inline config/webpacker.yml]: remove all the conflict lines, please.

Comment (starocketeer) [reply config/webpacker.yml]: Done

**Comment by starocketeer** [comment]
Ready

File: app/controllers/static_pages_controller.rb [hunk 21]
```diff
@@ -0,0 +1,5 @@
+class StaticPagesController < ApplicationController
+    def home; end
+        
+    def help; end
```
Comment (saobv-2223) [inline app/controllers/static_pages_controller.rb]:
```suggestion
  def home; end
        
  def help; end
```

File: bebel.config.js [hunk 22]
```diff
@@ -0,0 +1,73 @@
...
+
```
Comment (saobv-2223) [inline bebel.config.js:74]:
```suggestion
  
  
```

File: config/settings.yml [hunk 23]
```diff
@@ -0,0 +1,6 @@
...
+
```
Comment (saobv-2223) [inline config/settings.yml:6]:
```suggestion
    home_page: http://www.railstutorial.org/
 
```

**Review by saobv-2223 — COMMENTED** [review]
fix comment and merge

File: app/controllers/static_pages_controller.rb (hunk 21 above)
Comment (starocketeer) [reply app/controllers/static_pages_controller.rb]: done

File: config/settings.yml (hunk 23 above)
Comment (starocketeer) [reply config/settings.yml:6]: done

**Pull request merged**